set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -217,55 +207,44 @@
 	@RequestMapping(value = {"/set/{identifier}"}, method = RequestMethod.PUT, 
 			produces = {HttpHeaders.CONTENT_TYPE_JSONLD_UTF8, HttpHeaders.CONTENT_TYPE_JSON_UTF8})
 	@ApiOperation(notes = SwaggerConstants.UPDATE_SAMPLES_JSONLD, value = "Update an existing user set", nickname = "update", response = java.lang.Void.class)
-	public ResponseEntity<String> updateUserSet(@RequestParam(value = WebUserSetFields.PARAM_WSKEY, required = false) String wskey,
+	public ResponseEntity<String> updateUserSet(
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_SET_ID) String identifier,
 			@RequestBody String userSet,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,
 			@RequestParam(value = WebUserSetFields.PROFILE, required = false, defaultValue = WebUserSetFields.PROFILE_MINIMAL) String profile,			
 			HttpServletRequest request
 			) throws HttpException {
 		
 		String action = "put:/set/{identifier}";
-		return updateUserSet(request, wskey, identifier, userSet, userToken, profile, action);
+		return updateUserSet(request, identifier, userSet, profile, action);
```

Comment:
the verifyWriteAccess method needs to be invoked here

Reply:
verifyWriteAccess invoked as suggested